### PR TITLE
blob: add metrics for bytes read and written

### LINKED ISSUE #1229: blob: add metrics for bytes read/written
Add metrics to `Read` and `Write` to capture the number of bytes read and written.

### PR DESCRIPTION
**Pull request by jba**
Fixes #1229.

### CODE REVIEW

File: blob/blob.go [hunk 1]
```diff
@@ -66,6 +66,10 @@
 //   are included because they call NewRangeReader.)
 // - NewWriter, from creation until the call to Close.
 //
+// It also collects the following metrics:
```
Comment (vangent) [inline blob/blob.go]: Not part of this PR, but should the section above say what the name of the metric is for each method?

File: blob/blob.go [hunk 2]
```diff
@@ -99,14 +106,17 @@ import (
...
+	b        driver.Bucket
+	r        driver.Reader
+	end      func(error) // called at Close to finish trace and metric collection
+	provider string      // for metric collection
 }
 
 // Read implements io.Reader (https://golang.org/pkg/io/#Reader).
 func (r *Reader) Read(p []byte) (int, error) {
 	n, err := r.r.Read(p)
+	stats.RecordWithTags(context.Background(), []tag.Mutator{tag.Upsert(oc.ProviderKey, r.provider)},
```
Comment (vangent) [inline blob/blob.go]: WDYT about keeping a counter here and exporting the metric in `Close` instead?

Comment (jba) [reply blob/blob.go]: That's less accurate (Close might not be called; or the reading might go on for minutes or hours) and less maintainable. Are you concerned about performance? Nothing is getting "exported" (sent off-box) at this point.  This line shouldn't take much time. Unless someone is foolishly reading in very small chunks, it's unlikely to be noticeable. But we could benchmark (against a real service with real buffer sizes).

Comment (vangent) [reply blob/blob.go]: OK, your arguments are convincing :-).

File: blob/blob.go (hunk 1 above)
Comment (jba) [reply blob/blob.go]: I just did that for pubsub in #1232. I kept the wording generic so we can cut and paste. If you like that, I'll repeat it here and elsewhere.
